sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -184,8 +184,10 @@ private[sql] object PartitioningUtils {
         return (None, None)
       }
 
-      if (basePaths.contains(currentPath)) {
+      if (basePaths.contains(currentPath) ||
+        basePaths.exists(_.toString.startsWith(currentPath.toString))) {
```

Comment:
I see. We are trying to check if there is a basePath starts with the currentPath. 

So, the actual problem is that `basePaths` in HDFSFileCatalog contains files, right? I discussed it with @tdas. He will have a pr to change `basePaths`. Let's review his fix together. What do you think?